### ISSUE TITLE
List of insertable symbols

### ISSUE THREAD

**Issue by cool9333** [mb1986/rm-hacks#144]
I would like to see a group of insertable symbols to put in documents that I could also customize (X, checkmark, question mark, initials, signature, etc.). Maybe a list that pops up from a long press of the pen tip to the screen and/or as a button click function in the Lamy hacks.

**Comment by PepikVaio** [comment]
What do you mean?
It would have to be curves? So they can be copied and used as classic notes.
That would be cool, but isn't that too much?
Could it be done?

**Comment by cool9333** [comment]
I was thinking of something like the stamps palette in Acrobat, except for
rM2. Some standard symbols but you could also add customer ones like a
signature or initials.

LKW

**Comment by PepikVaio** [comment]
Yes, I understand that but how would rm work with that?
Or what format would it be in, where would it be stored?

What format would it be?
Because you can't extract a pattern from a pdf.

**Comment by cool9333** [comment]
¯\_(ツ)_/¯


Just a noob making requests from the gurus. If it can't be done, no worries.

**Comment by PepikVaio** [comment]
Yeah, I get it, I'm thinking out loud again.
And maybe once @mb1986  reads this, he'll be clear.

**Comment by mb1986** [comment]
It may be the hard one.

**Comment by chucktomasi** [comment]
I had requested something like this from RM a couple years ago. I could see it as a way to draw a shape, icon, signature, or what ever. Copy it to the buffer. An icon on the aide tool ar could be available to save the strokes there or show existing ones to copy. 

Basically a fast way to copy/paste stuff you have done before only instaed of between docs & pages, thry are quickly accessible from the side panel.

And since they are stored as standard strokes/vectors, you can share them with others using RCU or raw files. 

Who wants the poop emoji or skull icons I have created?

**Comment by cool9333** [comment]
No worries, my friend. You already do incredible things with your mods;
don't waste time on this if it will eat up too much of your valuable time.
Just a pipe dream, not like I can't check a box or sign my name with my rM2
already. 😉

Thanks,
LKW

**Comment by cool9333** [comment]
I would like to take a look at your emojis.

LKW

**Comment by chucktomasi** [comment]
![20240107_135911](https://github.com/mb1986/rm-hacks/assets/17863775/e358edfa-d7e0-4c82-8b9b-0bc034c1be15)
 This is the front page where I store the frequent ones. The subsequent pages have basic components or larger versions (aka a workbench).

All are hand drawn. Some kind of suck, but I am no artist. Always tweaking these. 

The vertical list on the far right usually travels in a pack to be placed on the right side of each page in my PDF planner. Not how I was thinking of implementing a solution for rm-hacks, but the end result would be quick access to icons for the same purpose.

**Comment by mb1986** [comment]
You've given me some idea... I have to check how copy and paste works. Maybe I could alter the clipboard contents and allow to paste it. Don't know if possible but worth verifying.

**Comment by chucktomasi** [comment]
Yeah, it would have to involve the copy/paste buffer somehow. If you stored the icon on a page (or perhaps a "multi-cell" page with an index to say 'here are the coordinates for the first, second, third, etc.) Display an icon on the sidebar like the bookmarks that pops out and shows a list of icons. When you tap one, the contents get copied from the file to the copy/paste buffer. I leave the challenge of how to put stuff into the buffer w/o actually accessing the copy tool to you, the expert. As a long time (+40 years) developer, architect, and part-time designer, I'm happy to collaborate.

**Comment by PepikVaio** [comment]
> I had requested something like this from RM a couple years ago. I could see it as a way to draw a shape, icon, signature, or what ever. Copy it to the buffer. An icon on the aide tool ar could be available to save the strokes there or show existing ones to copy.
> 
> Basically a fast way to copy/paste stuff you have done before only instaed of between docs & pages, thry are quickly accessible from the side panel.
> 
> And since they are stored as standard strokes/vectors, you can share them with others using RCU or raw files.
> 
> Who wants the poop emoji or skull icons I have created

Very good idea with copy and paste :)

**Comment by knox-dawson** [comment]
Most unicode dingbats can be converted to lines using [drawj2d](https://sourceforge.net/p/drawj2d/wiki/Examples_reMarkable/), but they come out as outlines since they apparently aren't available in the "Lines" font that drawj2d can use.  They may look a bit choppy in the screenshot, but when you cut and paste and expand, they are very crisp.  Just mentioing this as an option for symbol creation, because I like the idea of a selection list.  On the other hand, I have pages like this in my ABC folder specifically to cut and paste elsewhere.

![dingbats](https://github.com/mb1986/rm-hacks/assets/154367947/3bba138d-3306-46d8-8042-a18611d698c2)

**Comment by PepikVaio** [comment]
I also have the site but to have it right in the dashboard would be awesome.
